### PR TITLE
Implement In-memory transcript

### LINKED ISSUE #11: Implement transcript in device side
Implement transcript in device code to connect implementation with other proof system implementations.

### PR DESCRIPTION
**Pull request by swjng**
This PR resolves #11 
PR #25 is closed due to memory issue, so we decided to implement transcript feature running with cpu.
## What does this PR change
- Add `CudaTranscript` to cpu implementation
    - Run with member functions of `CudaTranscript` such as `write_field_element`
    - can change field and hash function
- Running Keccak256 hash function to make challenges in this implementation

### CODE REVIEW

File: sumcheck/src/gpu/sumcheck.rs [hunk 1]
```diff
@@ -97,34 +111,14 @@ impl<F: PrimeField + FromFieldBinding<F> + ToFieldBinding<F>> GPUApiWrapper<F> {
...
-        let squeeze_challenge = self.gpu.get_func("sumcheck", "squeeze_challenge").unwrap();
-        let launch_config = LaunchConfig {
-            grid_dim: (1, 1, 1),
-            block_dim: (1, 1, 1),
-            shared_mem_bytes: 0,
-        };
-        unsafe {
-            squeeze_challenge.launch(launch_config, (challenges, round))?;
-        }
+        let fes = self
```
Comment (DoHoonKim8) [inline sumcheck/src/gpu/sumcheck.rs:118]: How about `round_evals` for clarity?

**Review by DoHoonKim8 — APPROVED** [review]
Nice bro

File: sumcheck/src/transcript.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+use sha3::Keccak256;
+use std::{
+    io::{Cursor, Read, Write},
+    marker::PhantomData,
+};
+
+pub type Keccak256Transcript<F> = CudaTranscript<Keccak256, F>;
+
+#[derive(Debug, Default)]
+pub struct CudaTranscript<H, F> {
```
[the remark this answers is not in the document]
Comment (jeong0982) [reply sumcheck/src/transcript.rs:17]: That's true, I will change it

File: sumcheck/src/gpu/sumcheck.rs (hunk 1 above)
Comment (jeong0982) [reply sumcheck/src/gpu/sumcheck.rs:118]: There is `round_evals` for argument, so `round_eval_values` may be okay I think

Comment (jeong0982) [reply sumcheck/src/gpu/sumcheck.rs]: I will merge this PR and if you think it is not appropriate then please let me know

**Pull request merged**